doc/storage.md

```diff
@@ -74,10 +53,78 @@ helm install -f <file> charts/piraeus-op
 ### After install
 
 On a cluster with the operator already configured (i.e. after `helm install`),
-you can edit the satelliteset configuration like this:
+you can edit the LinstorSatelliteSet configuration like this:
 
 ```
 $ kubectl edit LinstorSatelliteSet.piraeus.linbit.com <satellitesetname>
 ```
 
 The storage pool configuration can be updated like in the example above.
+
+## Preparing physical devices
+
+By default, LINSTOR expects the referenced VolumeGroups, ThinPools and so on to be present. You can use the
+`devicePaths: []` option to let LINSTOR automatically prepare devices for the pool. Eligible for automatic configuration
+are block devices that:
+
+* Are a root device (no partition)
+* do not contain partition information
+* have more than 1 GiB
+
+To enable automatic configuration of devices, set the `devicePaths` key on `storagePools` entries:
+
+```yaml
+  storagePools:
+    lvmPools:
+    - name: lvm-thick
+      volumeGroup: drbdpool
+      devicePaths:
+      - /dev/vdb
+    lvmThinPools:
+    - name: lvm-thin
+      thinVolume: thinpool
+      volumeGroup: linstor_thinpool
+      devicePaths:
+      - /dev/vdc
+      - /dev/vdd
+```
+
+Currently, this method supports creation of LVM and LVMTHIN storage pools.
+
+#### `lvmPools` configuration
+* `name` name of the LINSTOR storage pool. Required
+* `volumeGroup` name of the VG to create. Required
+* `devicePaths` devices to configure for this pool. Must be empty and >= 1GiB to be recognized. Optional
+* `raidLevel` LVM raid level. Optional
+* `vdo` Enable [VDO] (requires VDO tools in the satellite). Optional
+* `vdoLogicalSizeKib` Size of the created VG (expected to be bigger than the backing devices by using VDO). Optional
+* `vdoSlabSizeKib` Slab size for VDO. Optional
+
+[VDO]: https://www.redhat.com/en/blog/look-vdo-new-linux-compression-layer
+
+#### `lvmThinPools` configuration
+* `name` name of the LINSTOR storage pool. Required
+* `volumeGroup` VG to use for the thin pool. Note: if `devicePaths` is set, this will be set to `linstor_$THINVOLUME`, to stay compatible with LINSTOR created pools. Required
```

Comment:
It is a bit strange for this to be required but in certain circumstances to not have a choice about the value. Would it be possible to make it optional, defaulting to the `linstor_$THINVOLUME` value? Maybe we should complain if neither `volumeGroup` nor `devicePaths` is set.

Reply:
Sadly the volumeGroup parameter isn't new, so marking it as optional would be considered a breaking change for K8s APIs.

What we can do is require a kind of XOR configuration, so you either have to set `volumeGroup: ""` and use `devicePaths` _or_ you set the volumeGroup, but then you can't use devicePaths

Reply:
I don't think we should reject storage pools where `devicePaths` is used and `volumeGroup` is equal to `linstor_$THINVOLUME`. That's what you have on a second reconciliation run, right? When `devicePaths` is used, we could reject values for `volumeGroup` that are not `""` or `linstor_$THINVOLUME`, and document here that `""` is OK. Does that work?

Reply:
sounds like a plan!